plugins/pantheon-files-ctags/plugin.vala

```diff
@@ -328,180 +328,125 @@ public class Marlin.Plugins.CTags : Marlin.Plugins.Base {
+            var style_context = new Gtk.StyleContext ();
+
+            string css = """
+            .nohover:hover {
+                background: @bg_color;
+            }
+            .cross-fix {
+                margin: 0;
+                padding: 0;
+            }
+            """;
+
+            try {
+                css_provider.load_from_data (css, -1);
+                Gtk.StyleContext.add_provider_for_screen (Gdk.Screen.get_default (), css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+            } catch (GLib.Error e) {
+                warning ("Failed to parse css style : %s", e.message);
+            }
 
-            select.connect (() => {
-                has_focus = true;
-            });
+            var color_button_remove = new Gtk.Image.from_icon_name ("window-close-symbolic", Gtk.IconSize.MENU);
+            color_button_remove.width_request = 16;
+            color_button_remove.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+            color_button_remove.get_style_context ().add_class ("cross-fix");
+
+            var color_button_red = new ColorButton ("red");
+            var color_button_orange = new ColorButton ("orange");
+            var color_button_yellow = new ColorButton ("yellow");
+            var color_button_green = new ColorButton ("green");
+            var color_button_blue = new ColorButton ("blue");
+            var color_button_violet = new ColorButton ("purple");
+            var color_button_slate = new ColorButton ("slate");
+
+            var colorbox = new Gtk.Grid ();
+            colorbox.set_size_request (150, 10);
+            colorbox.set_column_spacing (9);
+            colorbox.margin_start = 3;
+            colorbox.halign = Gtk.Align.START;
+            colorbox.add (color_button_remove);
+            colorbox.add (color_button_red);
+            colorbox.add (color_button_orange);
+            colorbox.add (color_button_yellow);
+            colorbox.add (color_button_green);
+            colorbox.add (color_button_blue);
+            colorbox.add (color_button_violet);
+            colorbox.add (color_button_slate);
+
+            // Cannot use this for every button due to this being a MenuItem
+            button_press_event.connect (button_pressed_cb);
 
-            deselect.connect (() => {
-                has_focus = false;
-            });
+            add (colorbox);
+            // Remove pesky hover state coloring
+            get_style_context ().add_class ("nohover");
+            show_all ();
         }
 
         private bool button_pressed_cb (Gdk.EventButton event) {
             /* Determine whether a color button was clicked on */
-            int y0 = (get_allocated_height () - BUTTON_HEIGHT) /2;
-            int x0 = BUTTON_WIDTH + SPACING;
+            int y0 = (get_allocated_height () - 16) / 2;
 
-            if (event.y < y0 || event.y > y0 + BUTTON_HEIGHT) {
+            if (event.y < y0 || event.y > y0 + 16) {
```

Comment:
I don't think you want to get rid of these constants. It's better to have something that makes sense to a future maintainer than just a magic number that isn't clear where it comes from

Reply:
I'll add a block of comments below the opening of this method detailing these.
It's better than to assume they're intrinsic to the button and whatnot (HiDPI button isn't 16 in size, for instance)